test/langtools/tools/javac/classfiles/attributes/innerclasses/InnerClassesTestBase.java

```diff
@@ -290,8 +292,10 @@ private void test(String classToTest, TestCase test, String...skipClasses) {
                     privateConstructor = "private A" + count + "() {}";
                     syntheticClasses.add("new A" + count + "();");
                 }
+                String instField = innerClassType == ClassType.CLASS && innerMod.contains(Modifier.ABSTRACT) ?
+                                                "int f; " : ""; // inhibit PERMITS_VALUE
                 sb.append(toString(innerMod)).append(' ');
-                sb.append(String.format("%s A%d {%s}\n", innerClassType, count, privateConstructor));
+                sb.append(String.format("%s A%d { %s %s}\n", innerClassType, count, instField, privateConstructor));
                 Set<String> flags = getFlags(innerClassType, innerMod);
                 class2Flags.put("A" + count, flags);
```

Comment:
Hacks to suppress ACC_PERMITS_VALUE bit from being set thereby causing the test to "fail"